### ISSUE TITLE
"Error: WebGL context was lost." while zooming out

### ISSUE THREAD

**Issue by zhangzhen** [gosling-lang/gosling.js#562]
I came across this error twice since I use gosling. Besides, the page crashed more than two times while I was editing json specs in the gosling editor. What have been written got lost, which annoys me a bit.

Zhen

**Comment by sehilyi** [comment]
Thank you for reporting these!

Regarding the "WebGL context" issue, could you share what was the example that made the error? 

I also confronted this error message several times but have not yet found a way to fix it (#457, #230). Hopefully, I can fix it soon as it seems to be quite critical.

Regarding the crash, could you share your spec whenever you confront that? We made multiple updates on our Editor to fix this issue (e.g., https://github.com/gosling-lang/gosling.js/pull/211), but it looks like we still need to take care of some cases.

**Comment by zhangzhen** [comment]
> Regarding the "WebGL context" issue, could you share what was the example that made the error?

The spec for this example is posted on #569. Gosling seems to take up too much memory (~1 Gb) while running the spec. I ran into this issue in the case of plain dataset. I don't know if the same issue will occur when data is served by Higlass Server because I don't use Higlass Server yet.

Can data be loaded from a local file? This will speed up the loading time.

If the same data is used in more than one tracks, the same data will be loaded multiple times instead of once. Avoiding duplicate data loading can definitely improve performance.

**Comment by zhangzhen** [comment]
@sehilyi This bug is so critical and can cause users to ditch gosling. It occurs also when the data comes from higlass server. Any progress on that?

Zhen

**Comment by sehilyi** [comment]
@zhangzhen, I agree that this is critical. I was having some problems reproducing this issue. 

Let me investigate this issue in detail during the winter break (i.e., until the new year).

**Comment by zhangzhen** [comment]
@sehilyi have you already fixed this bug?

Zhen

**Comment by sehilyi** [comment]
Closing this since there is another GitHub Issue for the same issue (https://github.com/gosling-lang/gosling.js/issues/230).
